src/AgentContainer.H

```diff
@@ -293,8 +299,37 @@ public:
         return m_num_diseases;
     }
 
+    /* Since read_workerflow update worker status Unit-wise and not community-wise - Get total teacher needed for each unit */
+    inline const amrex::Gpu::DeviceVector<int>& getUnitTotalTeacherCounts() const {
+        return Unit_total_teacher_counts;
+    }
+    inline const amrex::Gpu::DeviceVector<int>& getUnitElem3TeacherCounts() const {
+        return Unit_elem3_teacher_counts;
+    }
+    inline const amrex::Gpu::DeviceVector<int>& getUnitElem4TeacherCounts() const {
+        return Unit_elem4_teacher_counts;
+    }
+    inline const amrex::Gpu::DeviceVector<int>& getUnitMidlTeacherCounts() const {
+        return Unit_midl_teacher_counts;
+    }
+    inline const amrex::Gpu::DeviceVector<int>& getUnitHighTeacherCounts() const {
+        return Unit_high_teacher_counts;
+    }
+    inline const amrex::Gpu::DeviceVector<int>& getUnitDaycrTeacherCounts() const {
+        return Unit_daycr_teacher_counts;
+    }
+
 protected:
 
+    amrex::iMultiFab student_counts;
+    amrex::iMultiFab teacher_counts;
+    amrex::Gpu::DeviceVector<int> Unit_total_teacher_counts;
+    amrex::Gpu::DeviceVector<int> Unit_elem3_teacher_counts;
+    amrex::Gpu::DeviceVector<int> Unit_elem4_teacher_counts;
+    amrex::Gpu::DeviceVector<int> Unit_midl_teacher_counts;
+    amrex::Gpu::DeviceVector<int> Unit_high_teacher_counts;
+    amrex::Gpu::DeviceVector<int> Unit_daycr_teacher_counts;
+
```

Comment:
Is it possible to have a 2D vector instead of separate variable for each teacher category, where one index will indicate category (total, elem, midl, high, etc) and the other will indicate unit?

Reply:
amrex::Gpu::DeviceVector<int>amrex::Gpu::DeviceVector<int> does not support 2D. Should I leave it or use flatten 1D vector?